src/main/kotlin/no/nav/tilleggsstonader/sak/behandlingsflyt/StegController.kt

```diff
@@ -18,12 +18,16 @@ class StegController(
     private val tilgangService: TilgangService,
 ) {
 
-    @PostMapping("behandling/{behandlingId}/inngangsvilkaar")
-    fun ferdigstillInngangsvilkår(@PathVariable behandlingId: UUID): UUID {
+    @PostMapping("behandling/{behandlingId}/ferdigstill")
+    fun ferdigstillSteg(@PathVariable behandlingId: UUID, @RequestBody request: FerdigstillStegRequest): UUID {
         tilgangService.validerTilgangTilBehandling(behandlingId, AuditLoggerEvent.UPDATE)
         tilgangService.validerHarSaksbehandlerrolle()
 
-        return stegService.håndterInngangsvilkår(behandlingId).id
+        return when (request.steg) {
+            StegType.INNGANGSVILKÅR -> stegService.håndterInngangsvilkår(behandlingId).id
+            StegType.VILKÅR -> stegService.håndterVilkår(behandlingId).id
+            else -> error("Steg $request.steg kan ikke ferdigstilles her")
+        }
```

Comment:
Hadde kanskje kunnet vurdert å flyttet dette til `stegService` for å ikke ha logikk i controllern

Reply:
Ja, enig